Feature/scree wake lock api (#23)
thinktecture/Thinktecture.Blazor: src/Thinktecture.Blazor.ScreenWakeLock/README.md

# Thinktecture.Blazor.ScreenWakeLock

[![NuGet Downloads (official NuGet)](https://img.shields.io/nuget/dt/Thinktecture.Blazor.ScreenWakeLock?label=NuGet%20Downloads)](https://www.nuget.org/packages/Thinktecture.Blazor.ScreenWakeLock/)

## Introduction

> A Blazor wrapper for the [ScreenWakeLock API](https://www.w3.org/TR/screen-wake-lock/).

Reviewer: ```suggestion
A Blazor wrapper for the [Screen Wake Lock API](https://www.w3.org/TR/screen-wake-lock/).
```

> The ScreenWakeLock API allows web applications to request a screen wake lock. Under the right conditions, and if allowed, the screen wake lock prevents the system from turning off a device's screen.

Reviewer: ```suggestion
The Screen Wake Lock API allows web applications to request a screen wake lock. Under the right conditions, and if allowed, the screen wake lock prevents the system from turning off a device's screen.
```

## Getting started

### Prerequisites

You need .NET 7.0 or newer to use this library.

[Download .NET 7](https://dotnet.microsoft.com/download/dotnet/7.0)

### Platform support

> [![Platform support for Web Share API](https://caniuse.bitsofco.de/image/wake-lock.png)](https://caniuse.com/?search=wake-lock)

Reviewer: ```suggestion
[![Platform support for Screen Wake Lock API](https://caniuse.bitsofco.de/image/wake-lock.png)](https://caniuse.com/?search=wake-lock)
```

### Installation

You can install the package via NuGet with the Package Manager in your IDE or alternatively using the command line:

```
dotnet add package Thinktecture.Blazor.ScreenWakeLock
```

## Usage

The package can be used in Blazor WebAssembly projects.

### Add to service collection

> To make the WebShareService available on all pages, register it at the IServiceCollection in `Program.cs` before the host is built:

Reviewer: ```suggestion
To make the ScreenWakeLockService available on all pages, register it at the IServiceCollection in `Program.cs` before the host is built:
```

```csharp
builder.Services.AddScreenWakeLockService();
```

### Checking for browser support

> Before using the ScreenWakeLock API, you should first test if the API is supported on the target platform by calling the `IsSupportedAsync()` method.

Reviewer: ```suggestion
Before using the Screen Wake Lock API, you should first test if the API is supported on the target platform by calling the `IsSupportedAsync()` method.
```

> This method returns a boolean to indicate whether the Web Share API is supported or not.

Reviewer: ```suggestion
This method returns a boolean to indicate whether the Screen Wake Lock API is supported or not.
```

```csharp
var isSupported = await screenWakeLock.IsSupportedAsync();
if (isSupported)
{

>     // enable share feature

Reviewer: ```suggestion
    // enable screen wake lock feature
```

}
else
{
    // use fallback mechanism or hide/disable feature
}
```

### Request screen wake lock

> To request a screen wake lock, you need to call the `RequestWakeLockAsync` method from the `IScreenWakeLockService` service.

Reviewer: ```suggestion
To request a screen wake lock, you need to call the `RequestWakeLockAsync()` method from the `IScreenWakeLockService` service.
```

It's a common practice to add `()` to method names to stress they are methods.

The browser can refuse the request for various reasons (for example, because the battery charge level is too low), 
so it's a good practice to wrap the call in a `try…catch` statement. 
The exception's message will contain more details in case of failure.

```csharp
try 
{
    await _screenWakeLockService.RequestWakeLockAsync();
}
catch(Exception e) 
{
    // Handle exxception
}
```

### Release screen wake lock

> You also need a way to release the screen wake lock, which is achieved by calling the `ReleaseWakeLockAsync` method of the `IScreenWakeLockService` service.

Reviewer: ```suggestion
You also need a way to release the screen wake lock, which is achieved by calling the `ReleaseWakeLockAsync()` method of the `IScreenWakeLockService` service.
```
